lib/spack/spack/spec.py

```diff
@@ -735,6 +740,13 @@ def canonical(self):
         return self.parent.dag_hash(), self.spec.dag_hash(), self.deptypes
 
 
+class CompilerFlag(str):
```

Comment:
Suggest adding a description (docstring) of the class / it's purpose since it is new.